plutus-pab/src/Plutus/PAB/Core/ContractInstance/STM.hs

```diff
@@ -149,16 +149,17 @@ data OpenTxOutProducedRequest =
 --   may be interested in.
 data BlockchainEnv =
     BlockchainEnv
-        { beCurrentSlot  :: TVar Slot -- ^ Current slot
-        , beTxChanges    :: TVar (UtxoIndex TxIdState) -- ^ Map holding metadata which determines the status of transactions.
-        , beTxOutChanges :: TVar (UtxoIndex TxOutBalance) -- ^ Map holding metadata which determines the status of transaction outputs.
-        , beCurrentBlock :: TVar BlockNumber -- ^ Current block
+        { beRollbackHistory :: Integer -- ^ How much history do we retain in the environment. Zero signifies no trimming is done.
```

Comment:
`Maybe Integer` instead of special casing `0` seems to me more the Haskelly thing to do here?

Reply:
Yes. I was considering doing that.